src/server.rs

```diff
@@ -394,11 +402,13 @@ async fn detection_content(
 
 async fn detect_context_documents(
     State(state): State<Arc<ServerState>>,
+    headers: HeaderMap,
     WithRejection(Json(request), _): WithRejection<Json<models::ContextDocsHttpRequest>, Error>,
 ) -> Result<impl IntoResponse, Error> {
     let request_id = Uuid::new_v4();
     request.validate()?;
-    let task = ContextDocsDetectionTask::new(request_id, request);
+    let filtered_headers = filter_header(&state.orchestrator.get_config(), headers).await;
+    let task = ContextDocsDetectionTask::new(request_id, request, filtered_headers);
```

Comment:
Update per other changes.

```suggestion
    let headers = filter_headers(&state.orchestrator.config().passthrough_headers, headers);
    let task = ContextDocsDetectionTask::new(request_id, request, headers);
```